### ISSUE TITLE
Patch view sometimes glitches out

### ISSUE THREAD

**Issue by dromer** [Polyterative/Patcher#121]
## Description

In some patches the view totally glitches out for a while before settling.

For instance this Neutron patch: https://patcher.xyz/patches/details/107
Might be because it's self-referencing?

Tested in both Firefox and Chromium.

(btw what is a "Sweep issue"?)

**Comment by Polyterative** [comment]
Sweep is a tool I used in the past. Removed from list of issues now.
Regarding the bug I need to lower the time to stabilize the graph, will update this issue

**Comment by Polyterative** [comment]
Implemented a fix and regression coverage for this.

Findings
- The graph layout engine (ForceAtlas2) was running too long with aggressive weighting, which made cyclic/self-referential patches visibly wobble before settling.
- Patch graph sizing was stateful across rebuilds (`sizeConstant` multiplied on each rebuild), so repeated refreshes could distort node/edge sizes and amplify visual instability.

Fixes pushed to `develop`
- 4de2e289 `fix(graph): stabilize patch details layout and add regression e2e`
- 532b0f80 `fix(graph): reduce patch graph jitter and deterministic sizing`

What changed
- ForceAtlas2 tuning in graph renderer:
  - reduced `edgeWeightInfluence` and adjusted slowdown/gravity
  - replaced fixed 10s run with bounded complexity-based runtime
  - ensured timeout cleanup on destroy
- Deterministic per-build size computation in patch graph:
  - removed cumulative `sizeConstant` mutation
  - compute + clamp size factor per rebuild
- Added Playwright regression E2E:
  - `e2e/patch-graph-stability.spec.ts`
  - currently targets known-stable route `/patches/details/5`

Validation
- Unit/focused tests: `40/40` passed
- Playwright regression against production base URL:
  - `BASE_URL=https://patcher.xyz npx playwright test e2e/patch-graph-stability.spec.ts --project=chromium --reporter=list`
  - `1 passed`

**Comment by dromer** [comment]
Nice. I assume it's not on the live page yet?

**Comment by Polyterative** [comment]
Live now! It's not a complete fix to the issue, but it should behave a bit better. 

> Nice. I assume it's not on the live page yet?

**Comment by dromer** [comment]
Yeah it still freaks a bit, but quickly settles to a fixed position :)
